pennylane/devices/default_clifford.py

```diff
@@ -466,50 +478,256 @@ def simulate(
+
+            # Recurse only if the simplification happened
+            if meas_obs_simp != meas_obs:
+                coeffs, paulis = self._convert_op_to_linear_comb(meas_obs_simp, coeffs, paulis)
+                return coeffs, paulis
+
+        # Case for Prod
+        if isinstance(meas_obs, Prod):
+            coeffs.append(1.0)
+            paulis.append(
+                (
+                    "".join([_GATE_OPERATIONS[op.name] for op in meas_obs.terms()[1][0]]),
+                    meas_obs.wires,
+                )
+            )
+            return coeffs, paulis
+
+        # Case for SProd
+        if isinstance(meas_obs, SProd):
+            cof, obs = meas_obs.terms()
+            gate_ops = obs
+            if len(gate_ops) == 1 and isinstance(gate_ops[0], Prod):
+                gate_ops = obs[0].terms()[1][0]
+            coeffs.extend(cof)
+            paulis.append(("".join([_GATE_OPERATIONS[op.name] for op in gate_ops]), meas_obs.wires))
+            return coeffs, paulis
+
+        # Add support for more case when the time is right
+        raise NotImplementedError(
+            f"default.clifford doesn't support expectation value calculation with {type(meas_obs)} at the moment."
+        )
+
+    def _measure_observable_sample(self, meas_op, stim_circuit, shots, sample_seed):
+        """Compute sample output from a stim circuit for a given Pauli observable"""
+        meas_dict = {"X": "MX", "Y": "MY", "Z": "MZ", "_": "M"}
+
+        coeffs, paulis = self._convert_op_to_linear_comb(meas_op, coeffs=[], paulis=[])
+        samples = []
+        for pauli, wire in paulis:
+            stim_circ = stim_circuit.copy()
+            for op, wr in zip(pauli, wire):
+                if op != "I":
+                    stim_circ.append(meas_dict[op], wr)
+            sampler = stim_circ.compile_sampler(seed=sample_seed)
+            samples.append(qml.math.array(sampler.sample(shots=shots), dtype=int))
+
+        return samples, qml.math.array(coeffs)
+
+    # pylint:disable=protected-access
+    def measure_statistical(self, circuit, stim_circuit, stim, seed=None):
+        """Given a circuit, compute samples and return the statistical measurement results."""
+        # Compute samples via circuits from tableau
+        num_shots = circuit.shots.total_shots
+        sample_seed = seed if isinstance(seed, int) else self._seed
+
         results = []
         for meas in circuit.measurements:
-            # Analytic case
-            if not circuit.shots:
-                # Computing density matrix via tableaus
-                if isinstance(meas, DensityMatrixMP):  # do first because it is a child of StateMP
-                    res = self._measure_density_matrix(tableau_simulator)
+            # Computing samples via stim's compiled sampler
+            if isinstance(meas, SampleMP):
+                results.append(
+                    self._measure_observable_sample(meas.obs, stim_circuit, num_shots, sample_seed)[
```

Comment:
do we only support sampling observables, eg. if a user does `qml.sample(qml.PauliZ(0))` as opposed to `qml.sample(wires=[0])`?

Reply:
Ahh, added it. :)